core/src/main/scala/org/apache/spark/deploy/history/ApplicationHistoryProvider.scala

```diff
@@ -62,4 +65,12 @@ private[history] abstract class ApplicationHistoryProvider {
    */
   def getConfig(): Map[String, String] = Map()
 
+  /**
+   * Writes out the event logs to the output stream provided. The logs will be compressed into a
```

Comment:
Isn't that an implementation detail?

Or, if you want to enforce a zip file as the output, shouldn't it be a `ZipOutputStream`? That actually sounds better since then it enforces a stronger contract regardless of the underlying provider implementation. Otherwise different providers may decide to return different types of files.